src/StreamMetrics.js

```diff
@@ -0,0 +1,282 @@
+
+                    this.report.storage.bytesWrittenPerSec = (metricsReport.metrics['broker/cassandra']) ? metricsReport.metrics['broker/cassandra'].writeBytes : 0
+                    this.report.storage.bytesReadPerSec = (metricsReport.metrics['broker/cassandra']) ? metricsReport.metrics['broker/cassandra'].readBytes : 0
+
+                    this.report.startTime = metricsReport.startTime
+                    this.report.currentTime = metricsReport.currentTime
+                    this.report.timestamp = metricsReport.currentTime
+                } else {
+                    // calculate averaged values
+                    this.report.broker.messagesToNetworkPerSec = throttledAvg(this.report.broker.messagesToNetworkPerSec, metricsReport.metrics['broker/publisher'].messages.rate)
+                    this.report.broker.bytesToNetworkPerSec = throttledAvg(this.report.broker.bytesToNetworkPerSec, metricsReport.metrics['broker/publisher'].bytes.rate)
+
+                    this.report.network.avgLatencyMs = throttledAvg(this.report.network.avgLatencyMs, metricsReport.metrics.node.latency.rate)
+                    this.report.network.bytesToPeersPerSec = throttledAvg(this.report.network.bytesToPeersPerSec, metricsReport.metrics.WebRtcEndpoint.outSpeed.rate || 0)
+                    this.report.network.bytesFromPeersPerSec = throttledAvg(this.report.network.bytesFromPeersPerSec, metricsReport.metrics.WebRtcEndpoint.inSpeed.rate || 0)
+                    this.report.network.connections = throttledAvg(this.report.network.connections, metricsReport.metrics.WebRtcEndpoint.connections || 0)
+
+                    if (metricsReport.metrics['broker/cassandra']) {
+                        this.report.storage.bytesWrittenPerSec = throttledAvg(this.report.storage.bytesWrittenPerSec, (metricsReport.metrics['broker/cassandra']) ? metricsReport.metrics['broker/cassandra'].writeBytes : 0)
+                        this.report.storage.bytesReadPerSec = throttledAvg(this.report.storage.bytesReadPerSec, (metricsReport.metrics['broker/cassandra']) ? metricsReport.metrics['broker/cassandra'].readBytes : 0)
+                    }
+
+                    this.report.currentTime = metricsReport.currentTime
+                    this.report.timestamp = metricsReport.currentTime
+                }
+
+                await this.publishReport()
+            } else {
+                const now = Date.now()
+                // calculations for min/hour/day
+                if (this.report.timestamp === 0) {
+                    // first iteration
+                    const lastMessages = await this.getResend(this.legacyStreamId, 1)
+
+                    if (lastMessages.length === 0) {
+                        await this.publishReport()
+                    } else {
+                        const messages = await this.getResend(this.streamIds.minuteStreamId, this.legacyInterval)
```

Comment:
Should the first argument be `this.legacyStreamId`?

Reply:
Btw if this is the case, is there a need to do 2 separate resend requests? You could just do
```
const messages = await this.getResend(this.streamIds.minuteStreamId, this.legacyInterval)
```

and handle the `messages.length === 0` case there and then?